porch/func/wrapper/wrapper.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+
+	pb "github.com/GoogleContainerTools/kpt/porch/func/evaluator"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog/v2"
+)
+
+var (
+	port       = flag.Int("port", 9446, "The server port")
+	entrypoint = flag.String("entrypoint", "function", "The entrypoint of the function")
+)
+
+func main() {
+	flag.Parse()
+
+	if err := run(); err != nil {
+		fmt.Fprintf(os.Stderr, "unexpected error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	address := fmt.Sprintf(":%d", *port)
+	lis, err := net.Listen("tcp", address)
+	if err != nil {
+		return fmt.Errorf("failed to listen: %w", err)
+	}
+
+	evaluator := &singleFunctionEvaluator{
+		entrypoint: *entrypoint,
+	}
+
+	klog.Infof("Listening on %s", address)
+
+	// Start the gRPC server
+	server := grpc.NewServer()
+	pb.RegisterFunctionEvaluatorServer(server, evaluator)
+	if err := server.Serve(lis); err != nil {
+		return fmt.Errorf("server failed: %w", err)
+	}
+	return nil
+}
+
+type singleFunctionEvaluator struct {
+	pb.UnimplementedFunctionEvaluatorServer
+
+	entrypoint string
+}
+
+func (e *singleFunctionEvaluator) EvaluateFunction(ctx context.Context, req *pb.EvaluateFunctionRequest) (*pb.EvaluateFunctionResponse, error) {
+	var stdout, stderr bytes.Buffer
+	cmd := exec.CommandContext(ctx, e.entrypoint)
```

Comment:
where does this code find the binary?
is it dynamically downloaded? I am not finding where/how that happens.

Reply:
It reads the OCI metadata to get the entrypoint now.